src/curves.rs

```diff
@@ -193,45 +205,253 @@ impl Curve for LineStringCurve {
+            geo::Closest::SinglePoint(closest_point) => {
+                let distance_along_curve = closest_point
+                    .haversine_distance(&self.geom.points().next().unwrap())
+                    + self.start_offset;
+
+                let begin = self.geom.coords().next().unwrap();
+                let end = self.geom.coords().next_back().unwrap();
+
+                let sign = match RobustKernel::orient2d(point.into(), *end, *begin) {
+                    Orientation::Clockwise => 1.,
+                    _ => -1.,
+                };
+                let offset = closest_point.haversine_distance(&point) * sign;
+
+                Ok(CurveProjection {
+                    distance_along_curve,
+                    offset,
+                })
+            }
+            geo::Closest::Intersection(_) => Err(CurveError::InvalidGeometry),
+            geo::Closest::Indeterminate => Err(CurveError::NotFiniteCoordinates),
+        }
+    }
+
+    fn resolve(&self, projection: CurveProjection) -> Result<Point, CurveError> {
+        let fraction = (projection.distance_along_curve - self.start_offset) / self.length;
+        if !(0. ..=1.).contains(&fraction) || fraction.is_nan() {
+            return Err(CurveError::NotOnTheCurve);
+        }
+
+        let total_length = self.length;
+        let fractional_length = total_length * fraction;
+        let mut accumulated_length = 0.;
+
+        // go through each segment and look for the one that frame the distance that we seek
+        for segment in self.geom.lines() {
+            let segment_length = segment.haversine_length();
+            if accumulated_length + segment_length >= fractional_length {
+                let segment_fraction = (fractional_length - accumulated_length) / segment_length;
+                let segment_start = Point::from(segment.start);
+                let segment_end = Point::from(segment.end);
+
+                // get the Point at a haversine distance between two Points
+                // of a certain fraction of length on this distance
+                return Ok(segment_start.haversine_intermediate(&segment_end, segment_fraction));
+            }
+            accumulated_length += segment_length;
+        }
+
+        Err(CurveError::NotOnTheCurve)
+    }
+
+    fn bbox(&self) -> Rect {
+        let min_point = geo::Point(self.geom.bounding_rect().unwrap().min());
+        let max_point = geo::Point(self.geom.bounding_rect().unwrap().max());
+
+        // add max_extend distance in South and then West direction
+        let min_point_extended = min_point
+            .haversine_destination(180., self.max_extent)
+            .haversine_destination(270., self.max_extent);
+
+        // add max_extend distance in North and then East direction
+        let max_point_extended = max_point
+            .haversine_destination(0., self.max_extent)
+            .haversine_destination(90., self.max_extent);
+
+        Rect::new(
```

Comment:
can’t you provide directly `Rect::new(min_point_extended, max_point_extended)` ? a `Point` should be `into` coord